Observer/NonBoltOrderObserver.php

```diff
@@ -131,11 +132,15 @@ public function execute(Observer $observer)
             $cart = $this->cartHelper->buildCartFromQuote($quote, $quote, $items, null, true, false);
             $customer = $quote->getCustomer();
             $storeId = $order->getStoreId();
-            $result = $this->createNonBoltOrder($cart, $customer, $storeId);
-            if ($result != 200) {
+            $result = $this->createNonBoltOrder($cart, $customer, $storeId, $paymentMethod);
+            $response = $result->getResponse();
+            if (empty($response)) {
                 $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
                 return;
             }
+
+            $payment->setAdditionalInformation("transaction_reference", $response->reference);
```

Comment:
Why are we setting "transaction_reference" to a payment instance that is not related to Bolt?
We could overwrite something essential for the particular method (payment gateway) to work correctly.

Reply:
Good point-- I will use `bolt_transaction_reference` so as to avoid overwriting anything.

We need the Bolt transaction reference to be set in order to ingest tracking information, which we want to do for all orders:

https://github.com/BoltApp/bolt-magento2/blob/master/Observer/TrackingSaveObserver.php#L119